custom/icds/repeaters/generators/phi.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+from django.core.serializers.json import DjangoJSONEncoder
+
+from corehq import toggles
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.repeater_generators import (
+    CaseRepeaterJsonPayloadGenerator,
+)
+
+
+class BasePayloadGenerator(CaseRepeaterJsonPayloadGenerator):
+    @staticmethod
+    def enabled_for_domain(domain):
+        return toggles.PHI_CAS_INTEGRATION.enabled(domain)
+
+
+class SearchByParamsPayloadGenerator(BasePayloadGenerator):
+    @staticmethod
+    def _gender(gender):
+        if gender:
+            if gender == 'male':
+                return 'M'
+            elif gender == 'female':
+                return 'F'
+        return ""
+
+    def get_payload(self, repeat_record, case):
+        data = {
+            "beneficaryname": case.name,
+            "fathername": case.get_case_property('fathers_name') or "",
+            "husbandname": case.get_case_property('husbands_name') or "",
+            "mothername": case.get_case_property('mothers_name') or "",
+            "gender": self._gender(case.get_case_property('gender')),
+            "villagecode": 442639,
+            "subdistrictcode": 3318,
+            "districtcode": 378,
+            "statecode": 22,
+            "dateofbirth": "",
+            "mobileno": case.get_case_property('mobile_number'),
+            "namelocal": "",
+            "mothernamelocal": "",
+            "fathernamelocal": "",
+            "husbandnamelocal": "",
+            "email": "",
+            "govt_id_name": "",
+            "govt_id_number": ""
+        }
```

Comment:
Oh good. My only reason behind this was because the API breaks if there is any field missing even if we don't cover it or has information about it. So fields like `govt_id_number` would never be available with us so being explicit about such fields might not be really useful here.